services/user-management/src/test/java/com/foodie/user/RoleControllerTest.java

```diff
@@ -0,0 +1,277 @@
+package com.foodie.user;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.UUID;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import com.foodie.models.users.Role;
+import com.foodie.service.RoleService;
+import com.github.javafaker.Faker;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@ExtendWith({ TestDataSetup.class })
+class RoleControllerTest {
+
+    @InjectMocks
+    private RoleController roleController;
+
+    @Mock
+    private RoleService roleService;
+
+    private final Faker faker = new Faker();
+
+    private Role createTestRole() {
+        return new Role(UUID.randomUUID(), faker.name().username(), faker.book().toString(), null, null, null);
+    }
+
+    private UUID createTestUUID() {
+        return UUID.randomUUID();
+    }
+
+    @Test
+    void contextLoads() {
+        assertNotNull(roleController);
+    }
+
+    // Create Role Test
+    @Test
+    void createRole_Success() {
+        Role role = createTestRole();
+
+        when(roleService.createRole(role)).thenReturn(role);
+
+        ResponseEntity<?> responseEntity = roleController.createRole(role);
+
+        assertEquals(HttpStatus.CREATED, responseEntity.getStatusCode());
+        assertEquals(Collections.singletonMap("data", role), responseEntity.getBody());
+        verify(roleService).createRole(role);
+    }
+
+    @Test
+    void createRole_BadRequest() {
+        Role role = createTestRole();
+
+        when(roleService.createRole(role)).thenReturn(null);
+
+        ResponseEntity<?> responseEntity = roleController.createRole(role);
+
+        assertEquals(HttpStatus.BAD_REQUEST, responseEntity.getStatusCode());
+        assertEquals(Collections.singletonMap("data", "Unable to process your request"), responseEntity.getBody());
+        verify(roleService).createRole(role);
+    }
+
+    @Test
+    void createRole_Exception() {
+        Role role = createTestRole();
+
+        when(roleService.createRole(role)).thenThrow(
+                new RuntimeException("{status: 400, message: 'An Error Occurred', timestamp: '2021-08-10T12:00:00'}"));
+
+        ResponseEntity<?> responseEntity = roleController.createRole(role);
+
+        assertEquals(HttpStatus.INTERNAL_SERVER_ERROR, responseEntity.getStatusCode());
+
+        String responseBody = responseEntity.getBody().toString();
```

Comment:
This could result in a null pointer exception

Reply:
Fixed.